netty/src/main/java/io/grpc/netty/AbstractNettyHandler.java

```diff
@@ -56,8 +58,12 @@
                        Http2Settings initialSettings) {
     super(decoder, encoder, initialSettings);
 
-    // TODO(nmittler): Use auto-refill once https://github.com/grpc/grpc-java/issues/1175 is fixed.
-    this.initialConnectionWindow = Integer.MAX_VALUE;
+    // Set the timeout for graceful shutdown.
+    gracefulShutdownTimeoutMillis(GRACEFUL_SHUTDOWN_TIMEOUT);
+
+    // Extract the connection window from the settings if it was set.
```

Comment:
This comment is a bit misleading, since the settings frame specifies the stream window, not connection window. You have to know we are copying from one to the other.

How about:
Extract the stream window from the settings if it was set, and use it for connection window